Developer branch

amroee2 opened the pull request.

amroee2 committed:
Refactor FlightRepositroy into multiple classes for easier testing

amroee2 committed:
Create IFlight interface and decouple flight class

amroee2 committed:
Refactor Booking and Passenger classes for easier testing

amroee2 committed:
Refactor FlightValidator for easier testing

amroee2 committed:
Refactor GeneralUtility and ManagerUtilties for easier testing

amroee2 committed:
Refactor Manager class and ManagerUtilities

Added new interfaces to make the testing easier, manager utilities was separated into two classes with interfaces, and updated error message handling

amroee2 committed:
Update general utility to reflect changes and fix bug in BookingChecker

amroee2 committed:
Add booking functionalities for manager into new classes and interfaces

amroee2 committed:
Add error logging functionalities for manager into new classes and interfaces

amroee2 committed:
Add file I/O functionalities for manager into new classes and interfaces

amroee2 committed:
Update ManagerUtilities to reflect previous changes

amroee2 committed:
Add Account functionalities for passenger into new classes and interfaces

amroee2 committed:
Add Booking functionalities for passenger into new classes and interfaces, organize files

amroee2 committed:
Add filight functionalities for passenger into new classes and interfaces

amroee2 committed:
Refactor classes and interfaces for easier testing

amroee2 committed:
Use Dependency Injection package for easier injection

amroee2 committed:
Fix infinite loop bugs in switch statements

amroee2 committed:
Write unit tests for validate and import

amroee2 committed:
Refactor log in method and add tests for PassengerAccount

amroee2 committed:
Refactor and add tests for booking a flight

amroee2 committed:
Refactor and Add unit tests for cancel and modifying a booking

amroee2 committed:
Refactor and Add tests for flight filtering

amroee2 committed:
Refactor and Add tests for Bookings filter

amroee2 committed:
Add unit tests for flight validation

amroee2 committed:
Fix common dependency bug

amroee2 committed:
Apply proper unit testing for Import all valid flights

Updated ShouldImportAllValidFlights() method to use proper Mocking (Callback)

amroee2 committed:
Fix test class namespace typo

hayatzsala commented on AirportTicketBookingSystem/Airport Repository/FlightExport.cs:
```diff
@@ -0,0 +1,32 @@
+
+namespace AirportTicketBookingSystem.Airport_Repository
+{
+    public class FlightExport : IFlightExport
```
class name should be name not verb, for example if it repository then add repository as suffix, it could convert it to name, "FlightExportRepository"

hayatzsala commented on AirportTicketBookingSystem/Airport Repository/FlightImport.cs:
```diff
@@ -0,0 +1,43 @@
+
+namespace AirportTicketBookingSystem.Airport_Repository
+{
+    public class FlightImport : IFlightImport
```
same here about class/interface naming

hayatzsala commented on AirportTicketBookingSystem/Models/IBooking.cs:
```diff
@@ -0,0 +1,27 @@
+﻿namespace AirportTicketBookingSystem.Models
+{
+    public interface IBooking
```
why you create interfaces for models?

hayatzsala commented on AirportTicketBookingSystem/Program.cs:
```diff
@@ -1,17 +1,47 @@
+            .AddScoped<IFlightExport, FlightExport>()
+            .AddScoped<IFlightImport, FlightImport>()
+            .AddScoped<IFlightValidator, FlightValidator>()
+            .AddScoped<IBooking, Booking>()
+            .AddScoped<IPassenger, Passenger>()
+            .AddScoped<IFlight, Flight>()
+            .AddScoped<IPassenger, Passenger>()
```
models is not service so no need for registering them in DI , then no need for interface for models

hayatzsala reviewed (commented):
Great JOB @amroee2 , Keep up the good work 👏 
I like how you divided and refactoring the code, and the test cases you handled
Please check the comments and let me know if you have any questions

amroee2 committed:
Fix Clean coding related issues

amroee2 committed:
Refactor flight import and export class names

amroee2 committed:
Refactor flight import method and test

Updated functionality to accept directory name and file name as parameters to make proper unit testing

amroee2 committed:
Fix issue with fixture default value

amroee2 committed:
Refactor tests to include console output

amroee2 committed:
Remove models interfaces
